### PR TITLE
Avoid open login redirect attacks via next

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- Addresses the following security advisory --> https://github.com/Clinical-Genomics/scout/security/advisories/GHSA-3x45-2m34-x95v

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. Test on main branch and this branch according to the instructions in the advisory

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by CR

### CODE REVIEW

**Comment by northwestwitch** [comment]
### Main branch:

image

image


### This branch

image

File: scout/server/app.py [hunk 1]
```diff
@@ -83,10 +83,7 @@ def check_user():
             relevant_endpoint = not (static_endpoint or public_endpoint)
             # if endpoint requires auth, check if user is authenticated
             if relevant_endpoint and not current_user.is_authenticated:
-                # combine visited URL (convert byte string query string to unicode!)
-                next_url = "{}?{}".format(request.path, request.query_string.decode())
-                login_url = url_for("public.index", next=next_url)
-                return redirect(login_url)
+                return redirect(url_for("public.index"))
```
Comment (dnil) [inline scout/server/app.py:86]: ✅

File: scout/server/blueprints/login/views.py [hunk 2]
```diff
@@ -130,7 +128,6 @@ def users():
 def perform_login(user_dict):
     if login_user(user_dict, remember=True):
         flash("you logged in as: {}".format(user_dict.name), "success")
-        next_url = session.pop("next_url", None)
-        return redirect(request.args.get("next") or next_url or url_for("cases.index"))
+        return redirect(url_for("cases.index"))
```
Comment (dnil) [inline scout/server/blueprints/login/views.py:131]: ✅

**Review by dnil — APPROVED** [review]
This eliminates the issue by disabling the forwarding feature. This is not a very well used feature anyway.

**Pull request merged**